### PR TITLE
silence warden expired session "errors"

### LINKED ISSUE #4698: silence warden expired session "errors"
``` Warden::NotAuthenticated: "kartones" user is not logged in```
```
"/app/controllers/application_controller.rb", line 43 in session_security_token_valid?
"/app/controllers/application_controller.rb", line 48 in validate_session
"/app/controllers/application_controller.rb", line 266 in current_viewer
"/app/controllers/carto/api/users_controller.rb", line 85 in session_user
"/app/controllers/carto/api/users_controller.rb", line 20 in get_authenticated_users
```

### PR DESCRIPTION
**Pull request by Kartones**
Closes #4698

### CODE REVIEW

**Comment by Kartones** [comment]
@ethervoid CR please

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -41,7 +41,9 @@ def update_session_security_token(user)
 
   def session_security_token_valid?(user)
     warden.session(user.username).key?(:sec_token) &&
-    warden.session(user.username)[:sec_token] == Digest::SHA1.hexdigest(user.crypted_password)
+      warden.session(user.username)[:sec_token] == Digest::SHA1.hexdigest(user.crypted_password)
+  rescue Warden::NotAuthenticated
+    false
```
Comment (ethervoid) [inline app/controllers/application_controller.rb]: I always prefer to add a return but is just a personal preference

**Comment by ethervoid** [comment]
:+1:

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Kartones) [inline app/controllers/application_controller.rb]: According to Ruby's style guide you shouldn't use return (https://github.com/Vizzuality/ruby-style-guide , `Avoid return where not required for flow of control`). I also dislike it a lot, but try to follow most Ruby rules...

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5392/))

**Pull request merged**